### PR TITLE
Avoid infinite waiting on tasks in tests

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

Open questions:
  * Do we need to make the timeout configurable?
  * How does this play with performance tests?

### CODE REVIEW

File: pulpcore/tests/functional/utils.py [hunk 1]
```diff
@@ -11,7 +11,8 @@ async def get_response(url):
         return await session.get(url)
 
 
-SLEEP_TIME = 0.3
+SLEEP_TIME = 0.5
+TASK_TIMEOUT = int(10 * 60 / SLEEP_TIME)  # 10 minutes
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/tests/functional/utils.py:15]: > Regarding the setting, it is configurable from utils.py, right?

I would not call this configurable. It is hard coded.
30 minutes is worth a try. We can maybe adjust later.

[the remark this answers is not in the document]
Comment (lubosmj) [reply pulpcore/tests/functional/utils.py]: I do not think it is worth creating a new option in conftest to make it REALLY configurable (https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.hookspec.pytest_addoption). I like it the way it is.

File: pulpcore/tests/functional/__init__.py [hunk 2]
```diff
@@ -798,6 +811,8 @@ def _monitor_task(task_href):
             if task.state in ["completed", "failed", "canceled"]:
```
Comment (ipanova) [inline pulpcore/tests/functional/__init__.py:811]: skipped state is missing

File: pulpcore/tests/functional/__init__.py [hunk 3]
```diff
@@ -785,8 +794,12 @@ def delete_orphans_pre(request, orphans_cleanup_api_client, monitor_task):
 
 @pytest.fixture(scope="session")
 def monitor_task(tasks_api_client, pulp_domain_enabled):
-    def _monitor_task(task_href):
-        while True:
+    def _monitor_task(task_href, timeout=None):
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/tests/functional/__init__.py:797]: It's timeout in seconds. The default is already precalculated to $\left\lfloor \frac{T}{\delta t} \right\rfloor$.
I'll add a docstring to the fixture.

File: pulpcore/tests/functional/__init__.py (hunk 2 above)
Comment (mdellweg) [reply pulpcore/tests/functional/__init__.py:811]: I can add it. TBH we never had a codepath that could lead to "skipped", as far as I remember.

File: pulpcore/tests/functional/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/tests/functional/__init__.py]: I also changed the default constant to measure seconds too.

File: pulpcore/tests/functional/__init__.py [hunk 4]
```diff
@@ -785,8 +794,17 @@ def delete_orphans_pre(request, orphans_cleanup_api_client, monitor_task):
...
-        while True:
+    """
+    Wait for a task to reach a final state.
+
+    Returns the task in "completed" state, or throws a `PulpTaskTimeoutError` in case the timeout
+    in seconds (defaulting to 30*60) exceeded or a `PulpTaskError` in case it reached any other
+    final state.
+    """
+
+    def _monitor_task(task_href, timeout=TASK_TIMEOUT):
```
Comment (ipanova) [inline pulpcore/tests/functional/__init__.py:805]: :+1:

File: pulpcore/tests/functional/__init__.py (hunk 2 above)
Comment (ipanova) [reply pulpcore/tests/functional/__init__.py]: i agree, for the sake of consistency 'skipped' is considered among final states so let's have it in there.

**Review by lubosmj — APPROVED** [review]

**Pull request merged**